src/sentry/static/sentry/app/views/dashboardsV2/orgDashboards.tsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import {browserHistory} from 'react-router';
+import {Params} from 'react-router/lib/Router';
+import {Location} from 'history';
+
+import {Client} from 'app/api';
+import AsyncComponent from 'app/components/asyncComponent';
+import NotFound from 'app/components/errors/notFound';
+import LightWeightNoProjectMessage from 'app/components/lightWeightNoProjectMessage';
+import SentryDocumentTitle from 'app/components/sentryDocumentTitle';
+import {t} from 'app/locale';
+import {PageContent} from 'app/styles/organization';
+import {Organization} from 'app/types';
+
+import {PREBUILT_DASHBOARDS} from './data';
+import {DashboardListItem, OrgDashboard, OrgDashboardResponse} from './types';
+
+type OrgDashboardsChildrenProps = {
+  dashboard: DashboardListItem;
+  orgDashboards: DashboardListItem[];
+  reloadData: () => void;
+};
+
+type Props = {
+  api: Client;
+  location: Location;
+  params: Params;
+  organization: Organization;
+  children: (props: OrgDashboardsChildrenProps) => React.ReactNode;
+};
+
+type State = {
+  // endpoint response
+  orgDashboards: OrgDashboardResponse[] | null;
+} & AsyncComponent['state'];
+
+class OrgDashboards extends AsyncComponent<Props, State> {
+  state: State = {
+    // AsyncComponent state
+    loading: true,
+    reloading: false,
+    error: false,
+    errors: [],
+
+    // endpoint response
+    orgDashboards: [],
+  };
+
+  getEndpoints(): ReturnType<AsyncComponent['getEndpoints']> {
+    const {organization} = this.props;
+
+    const url = `/organizations/${organization.slug}/dashboards/`;
+
+    return [['orgDashboards', url]];
+  }
+
+  getOrgDashboards(): OrgDashboard[] {
+    const {orgDashboards} = this.state;
+
+    if (!Array.isArray(orgDashboards)) {
+      return [];
+    }
+
+    return orgDashboards.map(dashboard => {
+      return {
+        type: 'org',
+        ...dashboard,
+      };
+    });
+  }
+
+  getCurrentDashboard(): DashboardListItem | undefined {
+    const {params} = this.props;
+    const dashboardId = params.dashboardId as string | undefined;
+    const orgDashboards = this.getOrgDashboards();
+
+    if (typeof dashboardId === 'string') {
+      return orgDashboards.find(dashboard => {
+        return dashboard.id === dashboardId;
+      });
```

Comment:
What happens if we fail to find here? Should we return an error?

Reply:
In `renderBody()`, we return `<NotFound />` if `undefined` is returned from this function.